src/terrain/generation/worley_noise.cpp

```diff
@@ -0,0 +1,87 @@
+#include "worley_noise.hpp"
+
+namespace terrain {
+
+namespace generation {
+
+
+NoisePosition
+WorleyNoise::distance_(NoisePosition x, NoisePosition y, WorleyPoint point) {
+    return std::sqrt(
+        std::pow(x - point.x_position, 2) + std::pow(y - point.y_position, 2)
+    );
+}
+
+double
+WorleyNoise::get_noise(NoisePosition x, NoisePosition y) const {
+    NoiseTileIndex x_tile = x / tile_size_;
+    NoiseTileIndex y_tile = y / tile_size_;
+
+    auto worley_points = get_points_(x_tile, y_tile);
+
+    NoisePosition value = tile_size_ * 2;
+    for (const auto& point : worley_points) {
+        NoisePosition d = distance_(x, y, point);
+        if (d < value)
+            value = d;
+    }
+
+    return value;
+}
+
+std::set<WorleyPoint>
+WorleyNoise::get_points_(NoiseTileIndex x_t, NoiseTileIndex y_t) const {
```

Comment:
```suggestion
WorleyNoise::get_points_(NoiseTileIndex xt, NoiseTileIndex yt) const {
```

`_t` is only for types